src/material-experimental/mdc-list/list.scss

```diff
@@ -50,16 +50,22 @@
   $size: 40px;
   $margin-value: 72px - $mdc-list-side-padding - $size;
 
-  margin-left: 0;
-  margin-right: $margin-value;
   width: $size;
   height: $size;
   border-radius: 50%;
 
-  // `.mdc-list-item` added for extra specificity to override MDC's built in styles.
-  [dir='rtl'] .mdc-list-item & {
-    margin-left: $margin-value;
-    margin-right: 0;
+  // If the avatar is displayed in the graphic section of a list-item (i.e. at the start),
+  // we override the default "graphic" horizontal spacing that is designed for icons, to
+  // work for actual avatar images. The avatar by default is used within the `graphic` section,
+  // but could be also outside in case of a selection-list option with a checkbox at start.
```

Comment:
I don't totally follow this comment- what's the practical difference between what we support and what MDC supports here?

Reply:
MDC assumes that the list as a whole will either show avatars (larger images) or icons (smaller images). However, we allow avatar vs icon to be decided on a per-item basis. Therefore we can't rely on their styles which are something like `.mdc-list.mdc-list--avatar-list .mdc-list-item ... {}`. Instead we add similar styles of are own.

This part specifically is dealing with avatars that are marked as `.mdc-list-item__graphic`. MDC list items have 2 content slots for icons:
- `.mdc-list-item__graphic` is for an icon that comes before the text
- `.mdc-list-item__meta` if for an icon that comes after the text

Their styles targeted at `.mdc-list--avatar-list` would normally override these margins to look right for avatars rather than icons, but since we can't use those styles we're doing the same thing manually here.

Most of this is covered by various existing comments in the file, but I'll see if I can organize it to make it a bit more clear.